### ISSUE TITLE
[Bug] homepage should not be downloaded when checkver script is specified

### ISSUE THREAD

**Issue by kazzarin** [lukesampson/scoop#5704]
#### Current Behavior

When `checkver.script` is specified in a manifest and `checkver.url` is not, checkver defaults to requesting the `homepage` url before running the script. If that request errors, the script is never run and checkver fails.
#### Expected Behavior

When `checkver.script` is specified, `homepage` and `checkver.url` should be ignored and the script should run as expected.
#### Additional context/output

I changed the [homepage](https://github.com/kazzarin/bucket/blob/22fb2662e9143a1712b5d806abfc2b9947b65e60/bucket/oracle-graalvm.json#L4) for a manifest in my own bucket, which produced the following error:
```
> .\bin\checkver.ps1 oracle-graalvm
oracle-graalvm: The operation has timed out.
URL https://www.oracle.com/java/graalvm/ is not valid
```
Reverting the [change](https://github.com/kazzarin/bucket/blob/56732b8d74f4babfbd3400ff43829dd2b08ec36d/bucket/oracle-graalvm.json#L4) fixed it. The same behaviour was observed in my bucket's excavator, so I'm sure it's not a problem with my local machine. The wiki docs on the [script property](https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#properties-of-checkver) don't mention if it can be used with `checkver.url` or `homepage`, so I assume based on the manifests in the given examples that `checkver.script` is intended to be used exclusively when specified.

Unrelated, the site seems to timeout because of the link in scoop's user agent.
### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:** 7.3.9

**Additional software:** Git

#### Scoop Configuration

```json
{
    "aria2-enabled":  false,
    "aria2-warning-enabled":  false,
    "cat_style":  "plain",
    "debug":  true,
    "gh_token":  "redacted",
    "last_update":  "2023-10-27T22:05:56.4511315-04:00",
    "scoop_branch":  "develop",
    "scoop_repo":  "https://github.com/ScoopInstaller/Scoop",
    "show_update_log":  false,
    "use_external_7zip":  true
}

```

**Comment by z-Fng** [comment]
It seems to be a feature. Relates to:
- #5080

**Comment by notPlancha** [comment]
> It seems to be a feature. Relates to:
> - #5080

It makes sense url for the script is checked, it's homepage being checked that is the issue. Homepage as I see it is purely informative

**Comment by z-Fng** [comment]
I agree. But Scoop supported using the homepage as the default checkver.url long ago.

e.g. https://github.com/ScoopInstaller/Extras/blob/master/bucket/android-studio.json

So did #5080. To avoid impacting existing users, we should not break #5080.

And the key problem we are facing is that failing to download homepage/checkver.url prevents checkver.script from running.

So I think this PR is a more comprehensive fix:
- #6490

ps: For the current code structure, modifying it so that the webpage is not loaded unless the $page parameter is used in the script might be somewhat complicated.

**Comment by notPlancha** [comment]
agreed
